core/src/test/scala/unit/kafka/server/KafkaApisTest.scala

```diff
@@ -708,6 +710,55 @@ class KafkaApisTest {
     EasyMock.replay(groupCoordinator)
   }
 
+  @Test
+  def testReassignmentBytesOutMetricsUpdatedOnFetchRequest(): Unit = {
+    val leaderEpoch = 0
+    val tp0 = new TopicPartition("tp", 0)
+
+    val fetchData = Collections.singletonMap(tp0, new FetchRequest.PartitionData(0,0, Int.MaxValue, Optional.of(leaderEpoch)))
+    val (_, fetchFromFollower) = buildRequest(new FetchRequest.Builder(
+      ApiKeys.FETCH.oldestVersion(), ApiKeys.FETCH.latestVersion(), 1, 1000, 0, fetchData))
+
+    setupBasicMetadataCache(tp0.topic, numPartitions = 1)
+    val hw = 3
+
+    replicaManager.fetchMessages(anyLong, anyInt, anyInt, anyInt, anyBoolean,
+      anyObject[Seq[(TopicPartition, FetchRequest.PartitionData)]], anyObject[ReplicaQuota],
+      anyObject[Seq[(TopicPartition, FetchPartitionData)] => Unit](), anyObject[IsolationLevel],
+      anyObject[Option[ClientMetadata]])
+    expectLastCall[Unit].andAnswer(new IAnswer[Unit] {
+      def answer: Unit = {
+        val callback = getCurrentArguments.apply(7).asInstanceOf[(Seq[(TopicPartition, FetchPartitionData)] => Unit)]
+        val records = MemoryRecords.withRecords(CompressionType.NONE,
+          new SimpleRecord(1000, "foo".getBytes(StandardCharsets.UTF_8)))
+        callback(Seq(tp0 -> FetchPartitionData(Errors.NONE, hw, 0, records,
+          None, None, Option.empty)))
+      }
+    })
+
+    val fetchMetadata = new JFetchMetadata(0, 0)
+    val fetchContext = new FullFetchContext(time, new FetchSessionCache(1000, 100),
+      fetchMetadata, fetchData, false)
+    expect(fetchManager.newContext(anyObject[JFetchMetadata],
+      anyObject[util.Map[TopicPartition, FetchRequest.PartitionData]],
+      anyObject[util.List[TopicPartition]],
+      anyBoolean)).andReturn(fetchContext)
+
+    expect(replicaQuotaManager.record(anyLong()))
+    expect(replicaManager.getLogConfig(EasyMock.eq(tp0))).andReturn(None)
+
+    val partition: Partition = createNiceMock(classOf[Partition])
+    expect(partition.isAddingReplica(anyInt())).andReturn(true)
```

Comment:
Could we add a test which tests that we do not increment the reassignmentBytesOutPerSec when this returns false?